<launch>
	<arg name="cam_count" default = "2" />
	<arg name="frame_rate" default = "30" />
	<arg name="usb_mobile_app_enable" default= "true" />
	<arg name="web_mobile_app_enable" default= "true" />
	<arg name="traffic_light_annotator_app_enable" default="true" />
    <!-- <arg name="input_narrow_video" default="/media/fyp/sdCard/traffic_light_videos/27-02-2022_11-02_narrow_cam.mp4"/>
    <arg name="input_wide_video" default="/media/fyp/sdCard/traffic_light_videos/27-02-2022_11-02_wide_cam.mp4"/> -->

	<param name="cam_count" value="$(arg cam_count)"/>
	<param name="frame_rate" value="$(arg frame_rate)"/>
	<param name="usb_mobile_app_enable" value="$(arg usb_mobile_app_enable)" />
	<param name="web_mobile_app_enable" value="$(arg web_mobile_app_enable)" />
	<param name="traffic_light_annotator_app_enable" value="$(arg traffic_light_annotator_app_enable)" />
    <!-- <param name="input_narrow_video" value="$(arg input_narrow_video)"/>
    <param name="input_wide_video" value="$(arg input_wide_video)"/> -->

    <!-- <node pkg="traffic_light_detector"
	  	  type="image_feeder"
	  	  name="image_feeder"
	  	  output="screen"
	>
    </node> -->


	<node pkg="traffic_light_detector"
	  	  type="traffic.py"
	  	  name="traffic"
	  	  output="screen"
	>
    </node>

	<node  if="$(eval arg('usb_mobile_app_enable') or arg('web_mobile_app_enable'))"
		  pkg="traffic_light_detector"
	  	  type="mobile_app.py"
	  	  name="mobile_app_traffic_light_detection"
	  	  output="screen"
	>
    </node>

	<node if="$(eval traffic_light_annotator_app_enable == true)"
		  pkg="traffic_light_detector"
	  	  type="annotation_app.py"
	  	  name="annotation_web_app"
	  	  output="screen"
	>
    </node>


	<!-- <group if="$(eval cam_count == 2)">
        <node 	pkg="rviz"
				type="rviz"
				name="$(anon rviz)"
				output="screen"
				args="-d /home/fyp/catkin_ws/src/traffic_light_detector/rviz/dual_cam_rviz_config.rviz"

        ></node>
    </group> 
    <group unless="$(eval cam_count == 2)">
        <node 	pkg="rviz"
				type="rviz"
				name="$(anon rviz)"
				output="screen"
	            args="-d /home/fyp/catkin_ws/src/traffic_light_detector/rviz/single_cam_rviz_config.rviz"
        ></node>
    </group>  -->


	

</launch>
